python/pyspark/sql/tests/test_udtf.py

```diff
@@ -406,6 +406,35 @@ def terminate(self):
             ],
         )
 
+    def test_udtf_cleanup(self):
+        with tempfile.TemporaryDirectory() as d:
+            path = os.path.join(d, "file.txt")
+
+            @udtf(returnType="x: int")
+            class TestUDTF:
+                def __init__(self):
+                    self.path = path
+
+                def eval(self, x: int):
+                    raise Exception("eval error")
```

Comment:
can you add another test case with an exception from `terminate` as well? Then `cleanup` should still get called?